requirements.txt

```diff
@@ -9,6 +9,16 @@ pyyaml
 scons        # Works around a bug in scons on EL 8.
 tabulate
 wheel
+pylint
+numpy
```

Comment:
Does DAOS itself have a proper dependency on numpy? As far as I can tell, we only use numpy in some debugging / research files.

Reply:
There are some tests that import it so it gets added here.  The alternative is to suppress or tolerate the import warning in the test code.

Reply:
This was more of a general question outside the scope of this PR.
I don't think the _**build**_ depends on it. I only see these two files - both of which are not used anywhere
https://github.com/daos-stack/daos/blob/03ff92311b801ae443580f360b3b56bf5e8c758e/src/client/pydaos/raw/daos_io.py#L7-L16
https://github.com/daos-stack/daos/blob/03ff92311b801ae443580f360b3b56bf5e8c758e/src/tests/ftest/util/write_some_data.py#L10

Reply:
I wonder if we should remove those files then?  I'll look into that some more as this work progresses.

Reply:
Or... if we eventually want them to work, add `numpy` as a proper dependency so I can use all the goodies provided :)
As an argument... check out array slicing... (also works in higher dimensions)
```
>>> a = numpy.array([[1,2], [3,4], [5,6]])
>>>
>>> a
array([[1, 2],
       [3, 4],
       [5, 6]])
>>> a[:,0]
array([1, 3, 5])
>>> numpy.sum(a[:,0])
9
>>> a[:,0] += 1
>>> a
array([[2, 2],
       [4, 4],
       [6, 6]])
```